### PR TITLE
Add fat jars for gradle plugin

### PR DESCRIPTION
**Pull request by ilmat192**

### CODE REVIEW

File: samples/README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ built using either command line tools (via `build.sh` script presented in each s
 See `README.md` in sample directories to learn more about specific samples and the building process.
 
 **Note**: If the samples are built from a source tree (not from a distribution archive) the compiler and the gradle
-plugin built from the sources are used. So one must build the compiler by running `./gradlew cross_dist` from the 
+plugin built from the sources are used. So one must build them by running `./gradlew cross_dist` from the
```
Comment (olonho) [inline samples/README.md]: Will cross_dist install to local Maven?

File: samples/settings.gradle [hunk 2]
```diff
@@ -22,6 +22,5 @@ include ':uikit'
 include ':win32'
 include ':videoplayer'
 include ':workers'
+include ':serialization'
```
Comment (olonho) [inline samples/settings.gradle]: Are you sure we shall include dependency on serialization by default?

File: tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/konan/tasks/KonanGenerateCMakeTask.kt [hunk 3]
```diff
@@ -51,10 +51,10 @@ open class KonanGenerateCMakeTask : DefaultTask() {
     private val host = HostManager.host
 
     private fun generateCMakeLists(
```
Comment (olonho) [inline tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/konan/tasks/KonanGenerateCMakeTask.kt]: Do we need that/does it work?

File: samples/README.md (hunk 1 above)
Comment (ilmat192) [reply samples/README.md]: Yes

File: tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/konan/tasks/KonanGenerateCMakeTask.kt (hunk 3 above)
Comment (ilmat192) [reply tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/konan/tasks/KonanGenerateCMakeTask.kt]: It works but it doesn't look like we really need it now.

**Pull request merged**